lib/nmatrix/io/csv_reader.rb

```diff
@@ -0,0 +1,153 @@
+#--
```

Comment:
This is not very good library design.

When you put the code of an nmatrix plugin **INSIDE THE NMATRIX CORE** source, it being a plugin does not have any meaning. It will load along with the nmatrix core gem.

This csv_reader file should be here at all. If you do this, you'll be loading this file with the core `nmatrix` gem. It should inside a module called `NMatrix::IO` in the `lib/nmatrix/csv.rb` file.

Please read the guidelines throughly and make sure you understand everything before committing anything else. I hope you very well understand what plugins are and why they were adopted in the first place.